### ISSUE TITLE
MySQL 5.7 JSON ordering might affect change detection

### ISSUE THREAD

**Issue by MisatoTremor** [doctrine/doctrine2#7187]
This is not really an issue with  ORM itself, but can happen in conjunction with DBAL and MySQL 5.7

```JSON
{"foo": "bar", "baz": "bar"}
```

```JSON
{"baz": "bar", "foo": "bar"}
```

**Comment by Ocramius** [comment]
@MisatoTremor regardless about what MySQL does, we can't change anything on our side, because we'd destroy behavior that users might be relying upon.

The JSON type can sort keys before serializing, but that would be troublesome too.

If the only side-effect here is a repeated `UPDATE` query, then I'd rather leave this as a known issue (compared to introducing more awkwardness in JSON handling)

**Comment by MisatoTremor** [comment]
Yes, I thought so. My line of thinking was also that this could be noted in the docs where appropriate.

**Comment by Ocramius** [comment]
@MisatoTremor that would make a lot of sense. That affects also affects the `object` and `simple_array` types. The file to edit is probably https://github.com/doctrine/dbal/blob/03c0797f034be10759dff6e818a4d928872d97dc/docs/en/reference/types.rst

**Comment by MisatoTremor** [comment]
Thanks for the hads up @Ocramius. Shouldn't that just affect ``json`` and the (deprecated) ``json_array`` types as the other two use the CLOB declaration or am I overlooking something there?

**Comment by Ocramius** [comment]
@MisatoTremor anything that the DB may manipulate/optimise after persistence.
